[Build] Minimal build with the Core ML EP fails

### Describe the issue

Minimal or extended minimal builds with the Core ML Execution Provider fail because getting the model metadata in `coreml_execution_provider.cc` on L63 requires `Graph::GetModel()`, which is only available in full (non-minimal) builds.

This can be fixed by falling back to the existing default behaviour of using `model_hash` as the `user_provided_key` in minimal builds rather than reading it from the model metadata. This removes the dependency on `Graph::GetModel()` in the minimal path.

### Urgency

_No response_

### Target platform

iOS

### Build script

```
python tools/ci_build/build.py \
    --build_dir build/repro \
    --config MinSizeRel \
    --ios --use_xcode \
    --apple_sysroot iphoneos --osx_arch arm64 --apple_deploy_target 14.0 \
    --use_coreml \
    --minimal_build extended \
    --skip_tests --parallel \
    --cmake_extra_defines CMAKE_POLICY_VERSION_MINIMUM=3.5
```

### Error / output

```
coreml_execution_provider.cc:63:38: error: 
      no member named 'GetModel' in 'onnxruntime::Graph'
   63 |   const auto& metadata = main_graph->GetModel().MetaData();
      |                          ~~~~~~~~~~  ^
coreml_execution_provider.cc:115:12: error: 
      no matching function for call to 'CreateSupportedPartitions'
  115 |   result = utils::CreateSupportedPartitions(graph_viewer, is_node_supported, on_group_closed,
      |            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

### Visual Studio Version

_No response_

### GCC / Compiler Version

_No response_

--- a/include/onnxruntime/core/graph/graph.h
+++ b/include/onnxruntime/core/graph/graph.h
@@ -1455,10 +1455,6 @@ class Graph {  // NOLINT(clang-analyzer-optin.performance.Padding): preserve exi
     SetInputs(AsSpan(inputs));
   }
 
-  const Model& GetModel() const {
-    return owning_model_;
-  }
-
   const logging::Logger& GetLogger() const {
     return logger_;
   }
@@ -1476,6 +1472,10 @@ class Graph {  // NOLINT(clang-analyzer-optin.performance.Padding): preserve exi
 #endif  // !defined(ORT_MINIMAL_BUILD)
 
 #if !defined(ORT_MINIMAL_BUILD) || defined(ORT_EXTENDED_MINIMAL_BUILD)
+  const Model& GetModel() const {
+    return owning_model_;
+  }
+
   /** Sets the type of a NodeArg, replacing existing type/shape if any */
   void SetNodeArgType(NodeArg& arg, const ONNX_NAMESPACE::TypeProto& type_proto);
 

